src/hooks/isProjectAllowed.js

```diff
@@ -53,30 +54,43 @@ const checkOwner = context => {
 
   const inWhitelist = async project => {
     if (
-      (await isUserInProjectWhiteList(context.app, project.ownerAddress.toLowerCase())) ||
-      [TraceStatus.PROPOSED, CampaignStatus.PROPOSED].includes(project.status)
+      isRequestInternal(context) ||
+      (await isUserInProjectWhiteList(context.app, project.ownerAddress.toLowerCase()))
     ) {
       return;
     }
-
     throw new errors.BadRequest(
       `project owner address ${project.ownerAddress} is not in the whitelist`,
     );
   };
 
   if (Array.isArray(items)) {
-    items.forEach(inWhitelist);
+    // eslint-disable-next-line no-restricted-syntax
+    for (const item of items) {
+      // eslint-disable-next-line no-await-in-loop
+      await inWhitelist(item);
+    }
   } else {
-    inWhitelist(items);
+    await inWhitelist(items);
   }
   return context;
 };
+const checkTraceStatusIsProposed = context => {
```

Comment:
Campaign owner can create a trace himself, skipping `Proposed` state. So the first state will be `Pending`.

Reply:
@aminlatifi good point, I added this ability ( and wrote some test cases for that)